task3.js

```diff
@@ -0,0 +1,70 @@
+
+function remoteControl() {
+  var state = false;
+  var temperature = 26
+  var mode = ['cool', 'dry', 'hot', 'auto']
+  var count = 0
+  var currentMode = mode[0]
+
+  this.powerButton = function() {
+    if (state == false)
+      state = true
+    else
+      state = false 
+    return state;
+  };
+  this.up = function() {
+    return ++temperature;
+  };
+  this.down = function() {
+    return --temperature;
+  }
+  this.down = function() {
+    return --temperature;
+  }
+  this.switchMode = function() {
+    count++
+    currentMode = mode[count%4]  
+  }
+  this.showModeTemperature = function() {
+    console.log('Mode: ' + currentMode )
+    console.log('Temperature: ' + temperature )
+  }
+  this.setTimer = function(milisec) {
+    setTimeout(function(){ state = false
+      console.log('Turned off because of timer')
+    }, milisec);
+  }
+}
+
+// var counter = new remoteControl();
+// var counter1 = new remoteControl();
+
+// console.log( counter1.powerButton() ); // ?
+// counter1.setTimer(10000);
+// console.log( counter1.powerButton() ); // ?
```

Comment:
Would this turn off or on? If off, then what about following calls?

Reply:
this would turn on AC and then you would set a timer and after timer call the next 45 line would off the AC and after time completes that particular function would OFF the already off AC,